dashboard/config/libraries/PoemBot.interpreted.js

```diff
@@ -53,19 +123,94 @@ function randomWord() {
 }
 
 function addLine(line) {
-  poem.lines.push(line || '');
+  poem.lines.push(line || [blankText]);
 }
 
-function setTitle(title) {
-  poem.title = title;
+function setTitle(line) {
+  if (line) {
+    poem.title = line[0].value;
+  }
 }
 
-function setAuthor(author) {
-  poem.author = author;
+function setAuthor(line) {
+  if (line) {
+  	poem.author = line[0].value;
+  }
 }
 
 function showPoem() {
   isVisible = true;
 }
 
-other.push(poemBotDraw);
\ No newline at end of file
+// TODO: would it be possible to re-use the background/foreground effect code from dance party?
+function setBackgroundEffect(effectName) {
+  if (effectName === 'solid') {
+    backgroundEffect = function() {
+      background('pink');
+    };
+  } else if (effectName === 'rainbow') {
+    var hue = 0;
+    colorMode(HSL, 360);
+    backgroundEffect = function() {
+      background(hue, 200, 200);
+      hue = (hue + 1) % 360;
+    };
+  }
+}
+
+function setForegroundEffect(effectName) {
```

Comment:
these effects are lifted pretty directly from https://github.com/code-dot-org/dance-party/blob/staging/src/Effects.js
I'm sure we'll want more/different effects, but these are just here for now as proof-of-concept/prototype to facilitate level-writing.